src/idl_parser.cpp

```diff
@@ -598,22 +598,20 @@ CheckedError Parser::ParseType(Type &type) {
       return Error("nested vector types not supported (wrap in table first)");
     }
     if (token_ == ':') {
-      if (!IsScalar(subtype.base_type)) {
-        return Error(
-              "arrays of non-scalar types not supported");
-      }
       NEXT();
       if (token_ != kTokenIntegerConstant) {
         return Error(
               "length of fixed-length array must be an integer value");
       }
-      int64_t fixed_length = StringToInt(attribute_.c_str());
-      if (fixed_length < 1 || fixed_length > 0x7fff) {
+      int64_t fixed_length = 0;
```

Comment:
`int64_t fixed_length = 0;` - uint16_t ?

Reply:
The return value of `StringToNumber` is an `int64_t`. The check prevents an over-/underflow.

Reply:
https://github.com/google/flatbuffers/blob/bff7ffbc5130cd46caf33b76b4bb0593fcd15066/include/flatbuffers/util.h#L325

Reply:
D'oh missed it! Fixed.